src/components/Modal/Modal.tsx

```diff
@@ -1,161 +1,42 @@
-              type="button"
-              className={'modalButton--outlined'}
-              value="10"
-              onClick={(e) => this.handleChange(e, false)}
-            >
-              {' '}
-              $10
-            </button>
-            <button
-              type="button"
-              className={'modalButton--outlined'}
-              value="25"
-              onClick={(e) => this.handleChange(e, false)}
-            >
-              {' '}
-              $25
-            </button>
-            <button
-              type="button"
-              className={'modalButton--outlined'}
-              value="50"
-              onClick={(e) => this.handleChange(e, false)}
-            >
-              {' '}
-              $50
-            </button>
-            <button
-              type="button"
-              className={'modalButton--outlined'}
-              value="100"
-              onClick={(e) => this.handleChange(e, false)}
-            >
-              {' '}
-              $100
-            </button>
-          </div>
-          <label htmlFor="custom-amount">Or enter an amount </label> <br />
-          <input
-            name="custom-amount"
-            type="number"
-            className={classnames(styles.customAmt, 'modalInput--input')}
-            onChange={(e) => this.handleChange(e, true)}
-            value={this.state.customInput ? this.state.amount : ''}
-            placeholder="$"
-            min="5"
-          />
-          {this.state.amount < 5 && this.state.customInput ? (
-            <div className={styles.errorMessage}>
-              Please enter an amount greater than $5.00
-            </div>
-          ) : (
-            ''
-          )}
-        </div>
+export const Modal = (props: Props) => {
 
-        <button
-          type="button"
-          className={classnames(styles.nextBtn, 'modalButton--filled')}
-          onClick={this.showBillingsModal}
-          disabled={this.state.amount < 5}
-        >
-          {' '}
-          Next
-        </button>
+  const { modalView } = useModalPaymentState();
+  const dispatch = useModalPaymentDispatch();
 
-        <Elements stripe={stripePromise}>
-          <ModalPaymentBox
-            showPayModal={this.state.showBillModal}
-            hidePaymentModal={this.hideBillingsModal}
-            amount={this.state.amount}
-            purchaseType={this.props.purchaseType}
-            sellerId={this.props.sellerId}
-          />
-        </Elements>
-      </form>
-    );
+  const closeModal = (e: any) => {
+    e.preventDefault();
+    dispatch({ type: CLOSE_MODAL, payload: undefined })
   }
+  
+  return (
+    <div
+      id="donation-form"
+      className={classnames(styles.donationsContainer, 'modalForm--form')}
+      style={{ display: modalView > -1 ? 'block' : 'none' }}
+    >
+      <button className={'closeButton--close'} onClick={closeModal} > 
+        ×
+      </button>
+      {modalView === 0 && <ModalAmount {...props}/>}
```

Comment:
Does JavaScript have enums?

Reply:
TypeScript does

Reply:
I agree this should live as an enum somewhere